akka-persistence-typed-tests/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorRetentionSpec.scala

```diff
@@ -127,6 +126,73 @@ class EventSourcedBehaviorRetentionSpec
   val pidCounter = new AtomicInteger(0)
   private def nextPid(): PersistenceId = PersistenceId.ofUniqueId(s"c${pidCounter.incrementAndGet()}")
 
+  "EventSourcedBehavior with snapshot predicate" must {
+    "delete events after snapshot" in {
+      val eventProbe = TestProbe[Try[EventSourcedSignal]]()
+      val pid = nextPid()
+      val snapshotSignalProbe = TestProbe[WrappedSignal]()
+      val alwaysSnapshot: Behavior[Command] =
+        Behaviors.setup { ctx =>
+          counter(
+            ctx,
+            pid,
+            snapshotSignalProbe = Some(snapshotSignalProbe.ref),
+            eventSignalProbe = Some(eventProbe.ref))
+            .withRetention(RetentionCriteria.snapshotWhen[State, Event]((_, _, _) => true).withDeleteEventsOnSnapshot)
```

Comment:
Same error as above:

```
akka-persistence-typed-tests/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorRetentionSpec.scala:141:60: missing parameter type
[error]             .withRetention(RetentionCriteria.snapshotWhen((_, _, _) => true).withDeleteEventsOnSnapshot)
```

Reply:
hmm, this is unfortunate. There are no types on the RetentionCriteria base trait, ofc. Therefore the types of the snapshotWhen are unrelated to the types of the EventSourcedBehavior. For example this would be wrong, but compiles fine:
```
.withRetention(RetentionCriteria.snapshotWhen[String, Int]((_, _, _) => true).withDeleteEventsOnSnapshot)
```